### PR TITLE
Basic-ish refactoring for MySQL, async, & transactions

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

**Comment by kshakir** [comment]
Not sure the status of the original branch, but @geoffjentry these commits allow slicking to run on my local mysql instance.

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -1,8 +1,28 @@
...
-  driver = org.h2.Driver
-  connectionPool = disabled
-  keepAliveConnection = true
-}
\ No newline at end of file
+database {
+  config = h2mem1
+
+  h2mem1 = {
+    url = "jdbc:h2:mem:test1"
```
Comment (mcovarr) [inline src/main/resources/application.conf]: Is there a reason to use h2 in Thurloe and HSQLDB in Cromwell?

Comment (kshakir) [inline src/main/resources/application.conf]: Not sure. This was a PR into a PR refactoring to add MySQL support at the request of Jeff. Feel free to ask over on the original PR #7.

Comment (geoffjentry) [inline src/main/resources/application.conf]: Right. There was a request to try hard to make this done by end of sprint so @kshakir was just trying to move some stuff along while @cjllanwarne was away

**Comment by geoffjentry** [comment]
Thanks @kshakir!

**Pull request merged**